basic/src/test/java/org/jboss/hal/testsuite/test/configuration/ee/EETestCaseAbstract.java

```diff
@@ -0,0 +1,147 @@
+        } else {
+            navigation = new FinderNavigation(browser, StandaloneConfigEntryPoint.class);
+            navigation.addAddress(FinderNames.CONFIGURATION, FinderNames.SUBSYSTEMS)
+                    .addAddress(FinderNames.SUBSYSTEM, "EE");
+        }
+        navigation.selectRow().invoke(FinderNames.VIEW);
+        Application.waitUntilVisible();
+        //TODO: remove this after HAL-836 is resolved
+        By selector = ByJQuery.selector(".link-bar-first");
+        browser.findElement(selector).click();
+        Console.withBrowser(browser).waitUntilLoaded();
+        page.view("EE");
+        Console.withBrowser(browser).waitUntilLoaded();
+        //END OF REMOVE
+        page.switchTab("Services");
+    }
+
+    protected void editTextAndVerify(ResourceAddress address, String identifier, String attributeName, String value) throws IOException, InterruptedException {
+        ConfigFragment config = page.getConfigFragment();
+        config.edit().text(identifier, value);
+        config.save();
+        reloadIfRequiredAndWaitForRunning();
+        verifier.verifyAttribute(address, attributeName, value);
+    }
+
+    protected void editTextAndVerify(ResourceAddress address, String identifier, String attributeName) throws IOException, InterruptedException {
+        editTextAndVerify(address, identifier, attributeName, RandomStringUtils.randomAlphabetic(6));
+    }
+
+    protected void editCheckboxAndVerify(ResourceAddress address, String identifier, String attributeName, Boolean value) throws IOException, InterruptedException {
+        ConfigFragment config = page.getConfigFragment();
+        config.edit().checkbox(identifier, value);
+        config.save();
+        reloadIfRequiredAndWaitForRunning();
+        verifier.verifyAttribute(address, attributeName, value.toString());
+    }
+
+    public void selectOptionAndVerify(ResourceAddress address, String identifier, String attributeName, String value) throws IOException, InterruptedException {
+        ConfigFragment config = page.getConfigFragment();
+        config.edit().select(identifier, value);
+        config.save();
+        reloadIfRequiredAndWaitForRunning();
+        verifier.verifyAttribute(address, attributeName, value);
+    }
+
+    protected void reloadIfRequiredAndWaitForRunning() {
+        final int timeout = 60000;
+        if (ConfigUtils.isDomain()) {
+            new DomainManager(CliClientFactory.getClient()).reloadIfRequiredAndWaitUntilRunning(timeout);
+        } else {
+            CliClientFactory.getClient().reload();
+        }
+    }
+
+    protected Boolean isErrorShowedInForm() {
+        By selector = ByJQuery.selector("div.form-item-error-desc:visible");
+        return isElementVisible(selector);
+    }
+
+    protected Boolean isElementVisible(By selector) {
+        try {
+            Graphene.waitModel().withTimeout(5, TimeUnit.SECONDS).until().element(selector).is().visible();
```

Comment:
It's not suitable to call `withTimeout` method after `waitModel`since the latter one already set predefined timeout to 10s (see related `arquillian.xml` configuration). 
Instead of `Graphene.waitModel().withTimeout(5, TimeUnit.SECONDS)` you shoud simply use `Graphene.waitAjax()` to avoid hardcoded timeouts.
Anyway this method should be moved to UI abstraction.